libs/crossplatform/src/Windows/Utils.cpp

```diff
@@ -133,4 +136,29 @@ CrossPlatformUtils::REG_STATUS CrossPlatformUtils::GetLongPathRegStatus()
   }
   return REG_STATUS::DISABLED;
 }
+
+std::filesystem::perms CrossPlatformUtils::GetCurrentUmask() {
+  // Only way to get current umask is to change it, so revert the possible change directly.
+  unsigned int umask = ::umask(0);
```

Comment:
`umask` is deprecated according to https://learn.microsoft.com/en-us/cpp/c-runtime-library/reference/posix-umask?view=msvc-160. It recommends to use `_umask` or `_umask_s` instead. As this implementation is now Windows specific, I think we should follow the Microsoft recommendation.